src/ngraph/frontend/onnx_import/op/squeeze.cpp

```diff
@@ -35,26 +39,33 @@ namespace ngraph
                 NodeVector inputs{node.get_ng_inputs()};
                 auto data = inputs.at(0);
                 auto data_shape = data->get_shape();
-                auto axes = node.get_attribute_value<std::vector<uint64_t>>("axes", {});
+                auto axes = node.get_attribute_value<std::vector<std::size_t>>("axes", {});
+                AxisVector input_order{reshape::get_default_axis_vector(data_shape.size())};
 
+                // Default behaviour is to squeeze all single dimension axes.
                 if (axes.empty())
                 {
-                    for (auto index = 0; index < data_shape.size(); ++index)
+                    for (auto it = std::begin(data_shape); it != std::end(data_shape);)
                     {
-                        if (data_shape.at(index) == 1)
+                        if (*it == 1)
                         {
-                            axes.push_back(index);
+                            data_shape.erase(it);
+                            continue;
                         }
+                        ++it;
```

Comment:
It can't be moved inside `for()` loop statement since we don't want to increment it at every step. Look at above [lines](https://github.com/NervanaSystems/ngraph/blob/e42e45f8260748ee71a01f67067fe7b2f16182b3/src/ngraph/frontend/onnx_import/op/squeeze.cpp#L50-L54).  If we erase any element from container all the iterators are invalidated, and our `it` iterator would point to next element.

Reply:
Check the documentation of `erase` for `std::vector`, it says:
"Invalidates iterators and references at or after the point of the erase, including the end() iterator."
https://en.cppreference.com/w/cpp/container/vector/erase

Reply:
It won't work like this. See here: http://coliru.stacked-crooked.com/a/1947dd4d5d743ab8. I've changed loop structure as you have proposed as you said it was misleading.